utils/appliance/__init__.py

```diff
@@ -163,15 +165,42 @@ class IPAppliance(object):
             the openshift host so the framework can interact with the project.
         db_host: If the database is located somewhere else than on the appliance itself, specify
             the host here.
+        db_port: Database port.
+        ssh_port: SSH port.
     """
     _nav_steps = {}
 
     evmserverd = SystemdService.declare(unit_name='evmserverd')
     db = ApplianceDB.declare()
 
+    CONFIG_MAPPING = {
+        'base_url': 'address',
+        'browser_steal': 'browser_steal',
+        'container': 'container',
+        'pod': 'container',
+        'openshift_creds': 'openshift_creds',
+        'db_host': 'db_host',
+        'db_port': 'db_port',
+        'ssh_port': 'ssh_port',
+    }
+    CONFIG_NONGLOBAL = {'base_url'}
+
+    @property
+    def as_json(self):
+        """Dumps the arguments that can create this appliance as a JSON. None values are ignored."""
+        return json.dumps({
+            k: getattr(self, k)
+            for k in set(self.CONFIG_MAPPING.values())})
+
+    @classmethod
+    def from_json(cls, json_string):
+        return cls(**json.loads(json_string))
+
     def __init__(
             self, address=None, browser_steal=False, container=None, openshift_creds=None,
-            db_host=None):
+            db_host=None, db_port=None, ssh_port=None):
+        self.ssh_port = ssh_port or ports.SSH
```

Comment:
What is ports.SSH gets overridden later? There is still an option for that. We will get rid of that anyway I think later so all these will be defined right here as constants.